doc/tutorial/tutorial.md

```diff
@@ -1706,6 +1706,33 @@ to optimize data structures layout:
 extern type IObj<'A>
 ```
 
+### `#[has_side_effects]`
```

Comment:
These are not really tutorial items, they are highly technical.
The tutorial should only describe the core language, and should be ideally short and easy to read.

Reply:
I think there's nothing wrong about having advanced topics covered in the tutorial. Maybe we should break it up into several chapters for easier navigation and to have all the basic stuff in one smaller document.